### DISCUSSION TITLE
get async contex for shiki in vue component

### DISCUSSION THREAD

**Discussion by Alexufo** [vuejs/vitepress#3548] [Q&A]
I want to create syntax highlighting for a variable because it doesn't work by design (I am loading code samples from a files )
I want to use shiki and convert code sample to html as decribed here
https://github.com/vuejs/vitepress/issues/2315#issuecomment-1528779869

but shiki use promises and I cant return string from component.

**Comment by brc-dd** [comment]
Can you explain what you're trying to do? If you're using data loaders files, you can await the result there and return strings from there?
